app/controllers/works_controller.rb

```diff
@@ -267,14 +267,14 @@ def datacite
   def datacite_validate
     @errors = []
     @work = Work.find(params[:id])
-    datacite_xml = Nokogiri::XML(@work.to_xml)
-    schema_location = Rails.root.join("config", "schema")
-    Dir.chdir(schema_location) do
-      xsd = Nokogiri::XML::Schema(File.read("datacite_4_4.xsd"))
-      xsd.validate(datacite_xml).each do |error|
-        @errors << error
-      end
-    end
+    datacite_serialization = work.resource.datacite_serialization
+    datacite_serialization.valid?
```

Comment:
Nice, I am glad you found that the `Datacite` class has a way to execute this without us having to duplicate the code.